websockets/client.py

```diff
@@ -243,7 +244,8 @@ def handshake(self, wsuri, origin=None,
             for name, value in extra_headers:
                 set_header(name, value)
 
-        set_header('User-Agent', USER_AGENT)
+        if not is_header_set('User-Agent'):
```

Comment:
I believe `if 'User-Agent' in request_headers:` would do the job more efficiently.

https://docs.python.org/3/library/email.compat32-message.html#email.message.Message.__contains__

There's no need to allocate a dict, build it from `request_headers`, call its `keys()` method.

The `set_header` function was a cute API idea that I need to get rid of... ;-)

Reply:
Is this what we're after?

    from email.message import Message

    m = Message()
    for t in request_headers:
        m.add_header(*t)

    if 'User-Agent' not in m:
         set_header('User-Agent', USER_AGENT)

Reply:
Or should I just make request_headers a Message() in the first place and get rid of the lambdas ?

Reply:
Oops, sorry, we don't have a Message yet at this point, it's just a list of pairs. Leave it as is.

Reply:
Ok no problem!